routes/cohortRouter.js

```diff
@@ -0,0 +1,106 @@
+//Calling the various required modules for express, knex, router and the membersFix
+const { response } = require("express");
+const express = require("express");
+const knex = require("../client");
+const router = express.Router();
+const fix = require('./membersFix');
+
+//GET method for the base '/cohorts' page with a search bar to find cohort by name
+router.get('/', (request, response) => {
+    if (!request.query.name) {
+        knex('cohorts').orderBy('createdAt', 'desc')
+        .then(cohorts => {
+            response.render('cohorts', {cohorts: cohorts});
+        });
+    } else {
+        knex('cohorts').where('name', 'ilike', request.query.name).first()
+        .then(cohorts => {
+            if (cohorts) {
+                response.redirect(`/cohorts/${cohorts.id}`);
+            } else {
+                response.render('notFound');
+            }
+        });
+    }
+});
+
+//GET method for the '/cohorts/new' page
+router.get('/new', (request, response) => {
+    response.render('new');
+});
+
+//POST method for the '/cohorts' page from the '/cohorts/new' page
+router.post('/', (request, response) => {
+    //Makes sure the fields are populated, otherwise reloads the page
+    if (request.body.members && request.body.name && request.body.imageUrl) {
+        //Calls on a fix function from membersFix.js to solve any newline/empty space problems in the members list
+        let members = fix.fix(request.body.members)
+        knex('cohorts').insert({
+            imageUrl: request.body.imageUrl,
+            name: request.body.name,
+            members: members.toString()
+        }).returning('*').then(cohorts => {
+            response.redirect(`/cohorts/${cohorts[0].id}`);
+        });
+    } else {
```

Comment:
display an error message if you are redirecting a user to `/new` due to missing fields.